features/dd-sdk-android-trace/src/main/java/com/datadog/exec/CommonTaskExecutor.java

```diff
@@ -66,8 +66,14 @@ public <T> ScheduledFuture<?> scheduleAtFixedRate(
     } else {
       try {
         final PeriodicTask<T> periodicTask = new PeriodicTask<>(task, target);
+
+        // was scheduleAtFixedRate before, but Android Lint gives this:
+        // Error: Use of scheduleAtFixedRate is strongly discouraged because it can lead to
+        // unexpected behavior when Android processes become cached (tasks may unexpectedly
+        // execute hundreds or thousands of times in quick succession when a process
+        // changes from cached to uncached); prefer using scheduleWithFixedDelay [DiscouragedApi]
         final ScheduledFuture<?> future =
-            executorService.scheduleAtFixedRate(
+            executorService.scheduleWithFixedDelay(
```

Comment:
In this case should we change the function name at L58 since it's wrapper of it?

Reply:
I would keep it the same to avoid more code changes. Ideally they have the same behavior - some task executed periodically, just the tiny details are different - in the former case it will be scheduled even if previous task is not yet completed (I believe), while in the latter case it won't.